### PR TITLE
Minor formatting changes to debug kernel messages

### PR DESCRIPTION
**Pull request by DonJayamanne**
Some changes I had made locally when working on IW debugger & IPyWidgets

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -190,6 +190,24 @@
...
                 "icon": "$(clear-all)"
+            },
+            {
+                "command": "jupyter-kernelManager.viewKernelMessagesAsTree",
+                "title": "Tree View",
+                "category": "Notebook",
+                "icon": "$(list-tree)"
+            },
+            {
+                "command": "jupyter-kernelManager.viewKernelMessagesAsList",
```
Comment (DonJayamanne) [inline package.json:201]: Viewing the messages as a list view was important when working on IW debugger & IPyWidgets (to see the sequence of messages as they were sent/received)

File: package.json [hunk 2]
```diff
@@ -190,6 +190,24 @@
...
+                "title": "List View",
+                "category": "Notebook",
+                "icon": "$(list-flat)"
+            },
+            {
+                "command": "jupyter-kernelManager.kernelMessageCopy",
```
Comment (DonJayamanne) [inline package.json:207]: This was very useful when working on IPyWidgets (being able to copy text such as comm_id, msg_id)

File: src/kernelSpy/kernelMessageProvider.ts [hunk 3]
```diff
@@ -149,17 +204,24 @@ class MessageTreeItem extends TreeItem {
...
+                    this.description = `${debugReply.content.command}, ${debugReply.content.body['sourcePath']}`;
+                }
             }
 
             const debugEvent = data.msg as KernelMessage.IDebugEventMsg;
             if (
-                data.msg.header.msg_type === 'debug_request' &&
-                data.msg.channel === 'control' &&
+                data.msg.header.msg_type === 'debug_event' &&
+                data.msg.channel === 'iopub' &&
```
Comment (DonJayamanne) [inline src/kernelSpy/kernelMessageProvider.ts:224]: bug fix

File: src/kernelSpy/kernelMessageProvider.ts [hunk 4]
```diff
@@ -149,17 +204,24 @@ class MessageTreeItem extends TreeItem {
...
                 data.msg.channel === 'control' &&
                 debugReply.content.command
             ) {
                 this.description = `${debugReply.content.command} (success: ${debugReply.content.success}, seq: ${debugReply.content.seq})`;
+                if (
+                    debugReply.content.command === 'dumpCell' &&
+                    debugReply.content.body &&
+                    typeof debugReply.content.body['sourcePath'] === 'string'
+                ) {
+                    this.description = `${debugReply.content.command}, ${debugReply.content.body['sourcePath']}`;
```
Comment (DonJayamanne) [inline src/kernelSpy/kernelMessageProvider.ts:217]: This was very useful with IW debugger

File: src/kernelSpy/kernelMessageProvider.ts [hunk 5]
```diff
@@ -41,55 +41,72 @@ const requestIconsByMessageType = new Map<MessageType, string>([
...
+                this.tooltip = complete.content.code;
             }
             const inspect = data.msg as KernelMessage.IInspectRequestMsg;
             if (
                 data.msg.header.msg_type === 'inspect_request' &&
                 data.msg.channel === 'shell' &&
                 inspect.content.code
             ) {
-                this.description = inspect.content.code.split('\r\n').join('\\r\\n').split('\n').join('\\n');
+                this.description = getSingleLineValue(inspect.content.code);
```
Comment (DonJayamanne) [inline src/kernelSpy/kernelMessageProvider.ts:100]: Moved all of this copy pasted code into a single function

**Review by rchiodo — APPROVED** [review]

**Review by amunger — APPROVED** [review]

**Pull request merged**